src/nginx/grpc_server_call.cc

```diff
@@ -362,8 +362,9 @@ void NgxEspGrpcServerCall::OnHttpBlockReading(ngx_http_request_t *r) {
 
     ngx_esp_request_ctx_t *ctx = ngx_http_esp_ensure_module_ctx(r);
     if (ctx->grpc_upstream_cancel) {
-      // std::swap(grpc_upstream_cancel, ctx->grpc_upstream_cancel);
-      (*ctx->grpc_upstream_cancel)();
+      std::function<void()> *grpc_upstream_cancel =
+          ctx->grpc_upstream_cancel.release();
+      (*grpc_upstream_cancel)();
```

Comment:
no,  grpc_upstream_cancel is a pointer, it need to be deleted.  A better way is to use swap

std::unique_ptr<std::function<>>  cancel_func_ptr;
std::swap(ctx->grpc_upstream_cancel, cancel_func_ptr)

Reply:
Oops. You're right. I pushed the change.